lib/mongoose_push.ex

```diff
@@ -90,17 +91,8 @@ defmodule MongoosePush do
 
   defp maybe_log(:ok), do: :ok
 
-  defp maybe_log({:error, reason} = return_value) when is_atom(reason) do
-    Logger.warn(~s"Unable to complete push request due to #{reason}")
-    return_value
-  end
-
   defp maybe_log({:error, reason} = return_value) do
-    Logger.warn(
-      ~s"Unable to complete push request due to unknown error: " <>
-        ~s"#{inspect(reason)}"
-    )
-
+    Logger.warn(~s"Unable to complete push request due to #{inspect(reason)}")
```

Comment:
Since we know we return a tuple with 2 atoms, let's match on it and report both values in the log message separately, e.g.:
```suggestion
    {category, service_error} = reason
    Logger.warn(~s"Unable to complete push request due to service error: #{service_error} (category: #{category})")
```